### PR TITLE
Bplan archiving

### PR DESCRIPTION
**Pull request by 2e2a**
Show information text when the bplan project is over

### CODE REVIEW

File: meinberlin/apps/bplan/views.py [hunk 1]
```diff
@@ -21,6 +23,12 @@ class BplanStatementFormView(ProjectContextDispatcher,
     template_name = 'meinberlin_bplan/statement_create_form.html'
     success_url = reverse_lazy('meinberlin_bplan:statement-sent')
 
+    def dispatch(self, request, *args, **kwargs):
+        project = self.get_project(*args, **kwargs)
+        if project.has_finished:
+            return HttpResponseRedirect(reverse('meinberlin_bplan:finished'))
```
Comment (xi) [inline meinberlin/apps/bplan/views.py]: Why a redirect? I prefer showing the information text on the same URL.

File: meinberlin/apps/projects/views.py [hunk 2]
```diff
@@ -127,6 +127,9 @@ def get_queryset(self):
                 Q(participants__pk=self.request.user.pk) |
                 Q(organisation__initiators__pk=self.request.user.pk) |
                 Q(moderators__pk=self.request.user.pk)
+            ) & (
+                Q(is_archived=False) |
+                Q(externalproject__bplan=None)
```
Comment (xi) [inline meinberlin/apps/projects/views.py]: This code deserves a comment.

File: meinberlin/apps/bplan/views.py (hunk 1 above)
Comment (2e2a) [reply meinberlin/apps/bplan/views.py]: The `finished` view has different permissions from the statement submission view. With the redirect I was able to use the PermissionRequired mixin.

File: meinberlin/apps/projects/views.py (hunk 2 above)
Comment (kleingeist) [reply meinberlin/apps/projects/views.py]: i just looked over this again after my holidays and am not sure if it does really work is_archived is also set to True via a filter.
i would like to look into it again. so please don't merge yet

Comment (kleingeist) [reply meinberlin/apps/projects/views.py]: okay the expression is alright. all hail to the glory of databases :D

**Comment by kleingeist** [comment]
lgtm. 
it's weird that the default project detail page is shown prior to the "statement" phases start and that it is opened on the information tab while there is no information, but that has been the same as before. maybe we should address this in another PR.

**Pull request merged**